### PR TITLE
Controller changes to retrieve number of trips for a given user

### PR DESCRIPTION
**Pull request by AddySridhar95**
The controller function to retrieve user profile information now returns the groups a user is in. For each group we also get to know whether the user is the driver or a rider.

Also added route to update user profile.

In frontend (trip details) page, we have to call user profile route for the user to get the count of trips

### CODE REVIEW

File: controllers/cpuser.js [hunk 1]
```diff
@@ -79,13 +80,60 @@ module.exports = {
...
+   * Gets user profile details.
+   * Details include information like - first_name, last_name, email, age, phone_no, license,
+   * rating, CPGroups. For each CPGroup, information about whether the user has joined the group,
+   * is a driver/rider is also returned
+   */
   getUserProfile(req, res) {
     let cpuserid = req.query.cpuserid;
     return sequelize.transaction( (t) => {
-      return CPUser.findByPrimary(cpuserid)
+      return CPUser.findByPrimary(cpuserid, {
```
Comment (acsant) [inline controllers/cpuser.js]: Create a new route for user profile with trips.

File: controllers/cpuser.js [hunk 2]
```diff
@@ -79,13 +80,60 @@ module.exports = {
...
-  }
+  },
+
+  /*
+   * Should be allowed to update first_name, last_name, license and phone_no
+   *
+   * Mandatory information required: cpuserid
+   */
+  updateUserProfile(req, res) {
+    var updateObject = {};
```
Comment (acsant) [inline controllers/cpuser.js]: Rename this to something more meaningfull

File: routes/common.js [hunk 3]
```diff
@@ -36,6 +36,14 @@ module.exports = function (coPilotRoutes) {
   // Get user profile for some id
   coPilotRoutes.get('/user/profile', cpUserController.getUserProfile);
 
+  // Update user profile for some id. The cpuserid is part of the post request body.
+  coPilotRoutes.post('/user/profile', (req, res) => {
+    cpUserController.updateUserProfile(req, res).then((result) => {
+      console.log({result});
```
Comment (acsant) [inline routes/common.js]: Remove the console logs

File: controllers/cpuser.js (hunk 1 above)
Comment (AddySridhar95) [reply controllers/cpuser.js]: done

**Pull request merged**